packages/slonik/src/factories/createConnectionPool.ts

```diff
@@ -173,14 +176,13 @@ export const createConnectionPool = ({
 
         connections.splice(connections.indexOf(connection), 1);
 
-        const waitingClient = waitingClients.shift();
```

Comment:
Same thing here. We’re calling waitingClients.shift() but then if for some reason we’re below the minimum pool size we return early and the waiting client never gets a connection.

I removed the `return` after `addConnection()`, I don't see an obvious reason it means we need to return early but could be missing something